Replace the Mach thread-register walk in NtFlushProcessWriteBuffers with a
hardware memory barrier. Querying translated x86 register state under
Rosetta 2 stalls worker threads badly; Apple Silicon's TSO memory model
makes __sync_synchronize() sufficient here.

--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -7060,42 +7060,12 @@
 
 #ifdef __APPLE__
 
-static kern_return_t (*p_thread_get_register_pointer_values)( thread_t, uintptr_t*, size_t*, uintptr_t* );
-static pthread_once_t tgrpvs_init_once = PTHREAD_ONCE_INIT;
-
-static void tgrpvs_init(void)
-{
-    p_thread_get_register_pointer_values = dlsym( RTLD_DEFAULT, "thread_get_register_pointer_values" );
-    if (!p_thread_get_register_pointer_values)
-        FIXME( "thread_get_register_pointer_values not supported for NtFlushProcessWriteBuffers\n" );
-}
-
 /**********************************************************************
  *           NtFlushProcessWriteBuffers  (NTDLL.@)
  */
 NTSTATUS WINAPI NtFlushProcessWriteBuffers(void)
 {
-    /* Taken from https://github.com/dotnet/runtime/blob/7be37908e5a1cbb83b1062768c1649827eeaceaa/src/coreclr/pal/src/thread/process.cpp#L2799 */
-    mach_msg_type_number_t count, i;
-    thread_act_array_t threads;
-
-    pthread_once( &tgrpvs_init_once, tgrpvs_init );
-    if (!p_thread_get_register_pointer_values) return STATUS_SUCCESS;
-
-    /* Get references to all threads of this process */
-    if (task_threads( mach_task_self(), &threads, &count )) return STATUS_SUCCESS;
-
-    for (i = 0; i < count; i++)
-    {
-        uintptr_t reg_values[128];
-        size_t reg_count = ARRAY_SIZE( reg_values );
-        uintptr_t sp;
-
-        /* Request the thread's register pointer values to force the thread to go through a memory barrier */
-        p_thread_get_register_pointer_values( threads[i], &sp, &reg_count, reg_values );
-        mach_port_deallocate( mach_task_self(), threads[i] );
-    }
-    vm_deallocate( mach_task_self(), (vm_address_t)threads, count * sizeof(threads[0]) );
+    __sync_synchronize();
     return STATUS_SUCCESS;
 }
 
